src/commands/createNewProject/PythonProjectCreator.ts

```diff
@@ -0,0 +1,164 @@
+        await this.tryGetPythonAlias(PythonAlias.python, minReqVersion);
+        await this.tryGetPythonAlias(PythonAlias.python3, minReqVersion);
+        if (!this.pythonAlias) {
+            throw new Error(localize('pythonVersionRequired', 'Python {0} is required to create a Python Function project with Azure Functions Core Tools.', minReqVersion));
+        }
+    }
+
+    private async tryGetPythonAlias(pyAlias: PythonAlias, minReqVersion: string): Promise<void> {
+        try {
+            const pyVersion: string = (await cpUtils.executeCommand(ext.outputChannel, undefined /*default to cwd*/, `${pyAlias} --version`)).substring('Python '.length);
+            if (semver.gte(pyVersion, minReqVersion)) {
+                this.pythonAlias = pyAlias;
+            }
+        } catch {
+            // ignore error, not installed under this alias
+        }
+    }
+
+    private async setVirtualEnviornment(): Promise<void> {
+        const funcEnv: string = 'func_env';
+        let pythonVenvPath: string;
+        const newButton: MessageItem = { title: 'Create New' };
+        const existingButton: MessageItem = { title: 'Use Existing' };
+        const venvRequired: string = localize('venvRequired', 'You must be running in a Python virtual environment to create a Python Function project. Create a new one or use an existing?');
+        const input: MessageItem = await ext.ui.showWarningMessage(venvRequired, { modal: true }, newButton, existingButton);
+        if (input === newButton) {
+            // if there is no func_env, create one as it's required for Python functions
+            await cpUtils.executeCommand(ext.outputChannel, this.functionAppPath, this.pythonAlias, '-m', 'venv', 'func_env');
+            pythonVenvPath = funcEnv;
+        } else {
+            const options: OpenDialogOptions = {
+                canSelectFiles: false,
+                canSelectFolders: true,
+                canSelectMany: false,
+                defaultUri: workspace.workspaceFolders && workspace.workspaceFolders.length > 0 ? workspace.workspaceFolders[0].uri : undefined,
+                openLabel: localize('select', 'Select')
+            };
+            const venvFspath: string = (await ext.ui.showOpenDialog(options))[0].fsPath;
+            pythonVenvPath = path.relative(this.functionAppPath, venvFspath);
+        }
+        await updateWorkspaceSetting(this.pythonVenvPathSetting, pythonVenvPath, this.functionAppPath);
+    }
+
+    private async getVenvActivatePath(platform: Platform): Promise<string> {
+        const venvPath: string | undefined = getFuncExtensionSetting(this.pythonVenvPathSetting, this.functionAppPath);
+        if (venvPath) {
+            switch (platform) {
+                case Platform.Windows:
+                    return path.join('.', venvPath, 'Scripts', 'activate');
+                case Platform.MacOS:
+                default:
+                    // assuming OSX and Linux use the same path -- need to some testing
```

Comment:
Don't leave this comment in the code. Either test it before merging or file an issue on GitHub